### DISCUSSION TITLE
Wrap elements in custom component

### DISCUSSION THREAD

**Discussion by eggerand13** [JustFly1984/react-google-maps-api#2319] [Q&A]
Hello,

thank you very much for creating this awesome project. It works really well and makes using it in React a real breeze. However, I have one problem I can't seem to find a solution for. I want to wrap one or more component(s) in a custom component (so that I can have easier control over state) but it doesn't work.
For example:

When I place a Polygon in the map like this it will work just fine:

`
        <LoadScript
            googleMapsApiKey="XXXXXXXXXXXXXXXXXXXXXXXXXXXX"
        >
            <GoogleMap
                mapContainerStyle={containerStyle}
                zoom={10}
            >
<Polygon
  paths={geo.coordinates}
 />
            </GoogleMap>
        </LoadScript>
`

However as soon as I wrap this Polygon in a custom component, it will not be displayed on the map.

```
function CustomPolygon(props: ICustomPolygonProps) {

    const { paths } = props;

    return <Polygon
        paths={paths}
    />
}
```

```
<LoadScript
            googleMapsApiKey="XXXXXXXXXXXXXXXXXXXXXXXXXXXX"
        >
            <GoogleMap
                mapContainerStyle={containerStyle}
                zoom={10}
            >
<CustomPolygon
  paths={geo.coordinates}
 />
            </GoogleMap>
        </LoadScript>
```

Is it this generally not possible?

Thank you for any help.

**Comment by JustFly1984** [comment]
please provide minimal reproduction in codesandbox.io
